src/coreclr/jit/morph.cpp

```diff
@@ -12121,6 +12121,63 @@ GenTree* Compiler::fgMorphMultiOp(GenTreeMultiOp* multiOp)
                 }
                 break;
             }
+            case NI_SSE_And:
+            case NI_SSE2_And:
+            case NI_AVX_And:
+            case NI_AVX2_And:
+            {
+                // Transforms ~v1 & v2 to VectorXxx.AndNot(v1, v2)
+                GenTree* op1 = hw->Op(1);
+                GenTree* op2 = hw->Op(2);
+                if (!gtIsActiveCSE_Candidate(hw))
+                {
+                    if (op1->OperIs(GT_HWINTRINSIC))
+                    {
+                        NamedIntrinsic      ni       = NI_Illegal;
+                        GenTreeHWIntrinsic* inner_hw = op1->AsHWIntrinsic();
+                        switch (inner_hw->GetHWIntrinsicId())
+                        {
+                            case NI_SSE_Xor:
```

Comment:
Just looking for `xor` alone isn't sufficient.

You specifically need to look for `(x ^ AllBitsSet) & y` or `x & (y ^ AllBitsSet)`.

This optimization likewise needs to be done for `Arm64` and it needs to handle the same, but also handle `AdvSimd_Not`.